core/templates/pages/exploration-player-page/services/concept-card-manager.service.spec.ts

```diff
@@ -0,0 +1,293 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the Concept Card Manager service.
+ */
+
+import { EventEmitter } from '@angular/core';
+import { TestBed, fakeAsync, flush, tick } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { StateCard } from 'domain/state_card/state-card.model';
+import { PlayerPositionService } from 'pages/exploration-player-page/services/player-position.service';
+import { ConceptCardManagerService } from './concept-card-manager.service';
+import { ExplorationEngineService } from './exploration-engine.service';
+import { StateObjectFactory } from 'domain/state/StateObjectFactory';
+import { InteractionObjectFactory } from 'domain/exploration/InteractionObjectFactory';
+import { RecordedVoiceovers } from 'domain/exploration/recorded-voiceovers.model';
+import { WrittenTranslationsObjectFactory } from 'domain/exploration/WrittenTranslationsObjectFactory';
+import { AudioTranslationLanguageService } from './audio-translation-language.service';
+
+describe('HintsAndSolutionManager service', () => {
+let ccms: ConceptCardManagerService;
+let pps: PlayerPositionService;
+let ees: ExplorationEngineService;
+let stateObjectFactory: StateObjectFactory;
+let mockNewCardOpenedEmitter = new EventEmitter<StateCard>();
+let mockNewCardAvailableEmitter = new EventEmitter();
+let interactionObjectFactory: InteractionObjectFactory;
+let writtenTranslationsObjectFactory: WrittenTranslationsObjectFactory;
+let audioTranslationLanguageService: AudioTranslationLanguageService;
+
+const WAIT_BEFORE_REALLY_STUCK_MSEC: number = 12000;
+const WAIT_FOR_TOOLTIP_TO_BE_SHOWN_MSEC: number = 2000;
+const WAIT_FOR_CONCEPT_CARD_MSEC: number = 6000;
+
+beforeEach(fakeAsync(() => {
+    TestBed.configureTestingModule({
+        imports: [HttpClientTestingModule]
+    });
+    pps = TestBed.inject(PlayerPositionService);
+    ees = TestBed.inject(ExplorationEngineService);
+    stateObjectFactory = TestBed.inject(StateObjectFactory);
+    spyOn(pps, 'onNewCardAvailable').and.returnValue(
+        mockNewCardAvailableEmitter);
+    spyOn(pps, 'onNewCardOpened').and.returnValue(
+        mockNewCardOpenedEmitter);
+    ccms = TestBed.inject(ConceptCardManagerService);
+    interactionObjectFactory = TestBed.inject(InteractionObjectFactory);
+    writtenTranslationsObjectFactory = TestBed.inject(
+        WrittenTranslationsObjectFactory);
+    audioTranslationLanguageService = TestBed.inject(
+        AudioTranslationLanguageService);
+}));
+
+it('should show concept card icon at the right time', fakeAsync(() => {
+    // Case when no hints exist
+    spyOn(ccms, 'conceptCardForStateExists').and.returnValue(true);
+    ccms.hintsAvailable = 0;
+    expect(ccms.isConceptCardTooltipOpen()).toBe(false);
+    expect(ccms.isConceptCardViewable()).toBe(false);
+    expect(ccms.isConceptCardConsumed()).toBe(false);
+
+    ccms.reset();
+
+    // For releaseHint.
```

Comment:
Please expand on this, I can't tell what this means. Ditto below.

Reply:
Done.